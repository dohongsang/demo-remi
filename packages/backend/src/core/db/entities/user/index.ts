import { Column, Entity } from "typeorm";
import { BaseEntity } from "../base";

@Entity()
export class UserEntity extends BaseEntity {
  @Column()
  first_name: string;

  @Column()
  last_name: string;

  @Column()
  email: string;

  @Column()
  hash_password: string;
}
