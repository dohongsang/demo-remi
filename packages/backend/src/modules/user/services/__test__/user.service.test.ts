import { UserModel } from "../../models/user.model";
import { UserService } from "../user.service";

const userDao = {
  findOne: jest.fn(),
  insert: jest.fn(),
} as any;

it("UserService => Handle to test createUserProfile", async () => {
  const payload = new UserModel();
  const userService = new UserService(userDao);
  userDao.insert.mockResolvedValue();
  await expect(
    userService.createUserProfile(payload)
  ).resolves.toMatchSnapshot();
});

it("UserService => Handle to test createUserProfile", async () => {
  const userService = new UserService(userDao);
  userDao.findOne.mockResolvedValue();
  await expect(userService.findMe("id_123")).rejects.toMatchSnapshot();
});
