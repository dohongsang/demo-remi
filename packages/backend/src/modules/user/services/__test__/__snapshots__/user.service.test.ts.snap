// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserService => Handle to test createUserProfile 1`] = `undefined`;

exports[`UserService => Handle to test createUserProfile 2`] = `[TypeError: Cannot read properties of undefined (reading 'id')]`;
